### PR TITLE
Implement getUserInfo, updateUser and deleteUser

### LINKED ISSUE #21: Add the use of SQL (MySQL/Postgres) Database for user

### PR DESCRIPTION
**Pull request by lhy-hoyin**
## `getUserInfo` (READ)
- POST to `/test/getUserInfo`
- Provide `id` or `email`

## `updateUser` (UPDATE)
- POST to `/test/update`
- Provide `id` and (`username` and/or `password`)

## `deleteUser` (DELETE)
- POST to `/test/deregister`
- Provide `id`

Closes #21.

### CODE REVIEW

File: backend/server/services/userService.js [hunk 1]
```diff
@@ -46,36 +46,72 @@ const createUser = async (email, password, confirmPassword) => {
...
+};
+
+const getUserInfo = async (userId, email) => {
+  try {
+    if (!userId && !email) {
+      throw { status: 400, message: 'Need at least id or email' };
+    }
+
+    if (userId != null)
+         return userDatabase.getUserInfoById(userId);
```
Comment (venuslimm) [inline backend/server/services/userService.js:83]: I think this is overindented

**Review by venuslimm — APPROVED** [review]
Just small indentation issue but other than that lgtm

File: backend/server/services/userService.js (hunk 1 above)
Comment (lhy-hoyin) [reply backend/server/services/userService.js:83]: It is, but its to match with the line below.
image

**Comment by lhy-hoyin** [comment]
^ new commit for update user.

With this, CRUD on users are done.

**Review by venuslimm — APPROVED** [review]
LGTM 😄

**Pull request merged**